### PR TITLE
feat: Management API + CLI command to create a new chunk (rollover partition)

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/influxdata/influxdb_iox/issues/979

# Rationale:
Need a way to manually control the chunks on an IOx cluster

# Changes:
* Adds management API and CLI for rolling over the current chunk
* CLI looks like `./influxdb_iox database partition new-chunk <db_name> <partition_key>`, per #979 

# Not Included in this PR:
* The notion of adding a "closing" state to chunks in the mutable buffer (this PR just adds an API onto what exists)
* No changes to the code that actually does the rollover.
* The mutable buffer still has exactly one chunk in "open" and everything else in "closed" states

# Example

```
./influxdb_iox database partition new-chunk 26f7e5a4b7be365b_917b97a92e883afc '2021-03-15 12:00:00'`
Ok
```

```
./influxdb_iox database chunk list 26f7e5a4b7be365b_917b97a92e883afc
[
  {
    "partition_key": "2021-03-15 12:00:00",
    "id": 0,
    "storage": "ClosedMutableBuffer",
    "estimated_bytes": 2565313
  },
  {
    "partition_key": "2021-03-15 12:00:00",
    "id": 1,
    "storage": "OpenMutableBuffer",
    "estimated_bytes": 0
  },
]

```

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: tests/end_to_end_cases/management_api.rs [hunk 1]
```diff
@@ -391,3 +391,94 @@ async fn test_partition_get_error() {
...
+    assert_eq!(chunks.len(), 1, "Chunks: {:#?}", chunks);
+    let partition_key = "cpu";
+
+    // Rollover the a second chunk
+    management_client
+        .new_partition_chunk(&db_name, partition_key)
+        .await
+        .expect("new partition chunk");
+
+    // Load some more data and now expect that we have a second chunj
```
Comment (tustvold) [inline tests/end_to_end_cases/management_api.rs:425]: Typo

Comment (alamb) [reply tests/end_to_end_cases/management_api.rs:425]:
```suggestion
    // Load some more data and now expect that we have a second chunk
```

File: tests/end_to_end_cases/management_api.rs [hunk 2]
```diff
@@ -391,3 +391,94 @@ async fn test_partition_get_error() {
...
+async fn test_new_partition_chunk_error() {
+    let fixture = ServerFixture::create_shared().await;
+    let mut management_client = Client::new(fixture.grpc_channel());
+
+    let err = management_client
+        .new_partition_chunk("this database does not exist", "nor_does_this_partition")
+        .await
+        .expect_err("expected error");
+
+    assert_contains!(err.to_string(), "Database not found");
```
Comment (tustvold) [inline tests/end_to_end_cases/management_api.rs:483]: Why not just match the expected error?

Comment (alamb) [reply tests/end_to_end_cases/management_api.rs:483]: I prefer matching on the string representation of the error messages as a second check that the message makes sense. I realize that is somewhat of an opinion and I would be happy to change these tests to check the structured type if you prefer

**Review by tustvold — APPROVED** [review]

File: tests/end_to_end_cases/management_api.rs (hunk 2 above)
Comment (tustvold) [reply tests/end_to_end_cases/management_api.rs:483]: I personally don't view the error message as part of the public API, with this role instead fulfilled by the enum variants, so part of me feels it should be tested. But tbh I don't really mind either way :smile:

**Comment by alamb** [comment]
🤔  Something happened here. Will fix

**Pull request merged**